blog/on_retros/README.md

```diff
@@ -0,0 +1,272 @@
+
+When tickets are well-defined and small, you tend to make more accurate guesses for what work can be done each sprint (on some of my teams, we decided 5s were "too big" and always tried to break them down as much as possible without losing the "cohesion" of the problem they were solving. In general, we found that 5s tended to break down in to 6-10 points worth of 2 and 3 pointers).
+
+Similarly, well groomed tickets are easier to finish, so just by investing in grooming, you will see big gains in velocity and less rollover.
+
+### Bottlenecks in Codereview
+
+Codereview is a common bottleneck for new teams. You have to explain every decision you made when tackling a ticket, and convince someone who wasn't present for any of it that you made optimal decisions. They didn't see the things you tried that didn't work, or the conversations you had to narrow down solutions. And it only gets worse if the PRs are big. Reviewing someone elses' PR is more a or less a trust exercise. It's very difficult to just look at a diff and be like "yup I totally understand the context of all of this is and this is perfect."
+
+Often times, this "handoff" results in thrashing as a review spends a while looking at it, leaves a bunch of comments that may or may not actually be helpful, and then it goes back to the dev, back to the reviewer, etc. The fewer times that happens the better.
+
+And most engineers don't just wait for the codereview to get done (especially at organizations where this takes days or hours, instead of minutes), so they are working on something else and now there is the added difficultly of context switching and interruptions.
+
+If this sounds like you, I'd recommend experimenting with pair programming: the "continuous code review." Two or more people working on a PR together is guaranteed better than one person making the changes and another person reviewing it after. It's more efficient, there are no hand-offs, discussions happen continuously, and it often results in better code from the start. You also avoid fun situations where a junior engineer spends days/weeks (oof) working on a change and then a senior engineer rejects it out of hand because it wrecks the whole architecture and it needs to be completely redone. If they were pairing from the start, the junior engineer learns much earlier on not to go down that rabbit-hole and everyone is happier. It feels bad to be put in the position where you have to reject someone's hard work out of hand.
+
+If your organization change management process is ok with it, I'd suggest allowing for paired PRs to merge immediately after tests pass, though of course you should still circulate the PR to allow other's the opportunity to stay informed or make suggestions, even if they are after merging a PR. You can always revert it or do a follow-up if someone brings up something smart two hours later!
+
+### Bottlenecks in UAT/Testing/QA
```

Comment:
One thing we should explore, which I have spoken to teams on is the difference between the terms. I see much confusion because of the lack of these differences.